src/github.com/matrix-org/dendrite/clientapi/readers/sync.go

```diff
@@ -3,13 +3,22 @@ package readers
 import (
 	"net/http"
 
+	"github.com/matrix-org/dendrite/clientapi/auth"
 	"github.com/matrix-org/util"
 )
 
 // Sync implements /sync
 func Sync(req *http.Request) (interface{}, *util.HTTPError) {
 	logger := util.GetLogger(req.Context())
-	logger.Info("Doing stuff...")
+	userID, err := auth.VerifyAccessToken(req)
+	if err != nil {
+		return nil, &util.HTTPError{
+			Code: 403,
```

Comment:
I would very much like your thoughts on this. This has decoupled the error JSON and the HTTP status code. Should we be bundling them into one thing or?